### ISSUE TITLE
quantize_fx module not working on x86 machine for any torch vision model

### ISSUE THREAD

**Issue by choudhary-devang** [pytorch/pytorch#136511]
### 🐛 Describe the bug

When i am using the quantize_fx module with qengine =  "X86" on ICELAKE machine, for any of the torch vision model.
I have tried several models like resnet, mobilenet, alexnet, etc. in all the cases i am getting memory formatting error.
this is similar for both eager and compile mode.
# ERROR 
![image](https://github.com/user-attachments/assets/5dd1fd81-514f-4947-85ef-1f56bc162b32)

# Script
![Screenshot 2024-09-24 092503](https://github.com/user-attachments/assets/945e3802-c294-412a-8431-89d858f15e3d)

import torch
from torch.ao.quantization import get_default_qconfig_mapping
from torch.quantization.quantize_fx import prepare_fx, convert_fx
import torchvision.models as models
import time
qconfig_mapping = get_default_qconfig_mapping()

qengine = 'x86'
torch.backends.quantized.engine = qengine
qconfig_mapping = get_default_qconfig_mapping(qengine)
model_name = "resnet18"
model = models.__dict__[model_name](pretrained=True)
x = torch.randn((1, 3, 224, 224), dtype=torch.float)
model.eval()
prepared_model = prepare_fx(model, qconfig_mapping, example_inputs=x)
prepared_model(x)
quantized_model = convert_fx(prepared_model)
quantized_model = torch.compile(quantized_model)
with torch.no_grad():
    print("***************************************Compiled Model**********************************")
    print(quantized_model)
    st = time.time()
    quantized_model(*x)
    et = time.time()
    print(f"the average time required = {et-st}\n")

do i have to change the script to work because i have tried different memory format for this

# Blog which i was referring.
https://pytorch.org/blog/int8-quantization/

### Versions

[requirements.txt](https://github.com/user-attachments/files/17107715/requirements.txt)
i have attached the requirements.txt file which i was using to run.
accelerate==0.33.0
aiohappyeyeballs==2.4.0
aiohttp==3.10.5
aiosignal==1.3.1
asttokens==2.4.1
attrs==24.2.0
beautifulsoup4==4.12.3
boto3==1.35.11
botocore==1.35.11
bs4==0.0.2
certifi==2024.7.4
charset-normalizer==3.3.2
cloudpickle==3.0.0
comm==0.2.2
datasets==2.21.0
debugpy==1.8.5
decorator==5.1.1
dill==0.3.8
distro==1.9.0
evaluate==0.4.2
executing==2.1.0
filelock==3.15.4
frozenlist==1.4.1
fsspec==2024.6.1
huggingface-hub==0.24.2
idna==3.7
iniconfig==2.0.0
iopath==0.1.10
ipykernel==6.29.5
ipython==8.27.0
jedi==0.19.1
Jinja2==3.1.4
jmespath==1.0.1
joblib==1.4.2
jupyter_client==8.6.2
jupyter_core==5.7.2
libcst==1.4.0
llvmlite==0.43.0
MarkupSafe==2.1.5
matplotlib-inline==0.1.7
MonkeyType==23.3.0
mpmath==1.3.0
multidict==6.0.5
multiprocess==0.70.16
mypy-extensions==1.0.0
nest-asyncio==1.6.0
networkx==3.3
numba==0.60.0
numpy==1.23.5
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.20.5
nvidia-nvjitlink-cu12==12.5.82
nvidia-nvtx-cu12==12.1.105
opencv-python==4.10.0.84
packaging==24.1
pandas==2.2.2
parso==0.8.4
patch==1.16
pexpect==4.9.0
pillow==10.4.0
platformdirs==4.2.2
pluggy==1.5.0
portalocker==2.10.1
prompt_toolkit==3.0.47
psutil==6.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
py-cpuinfo==9.0.0
pyarrow==17.0.0
Pygments==2.18.0
pynvml==11.5.3
pytest==8.3.2
pytest-benchmark==4.0.0
python-dateutil==2.9.0.post0
pytz==2024.1
PyYAML==6.0.1
pyzmq==26.2.0
regex==2024.7.24
requests==2.32.3
s3transfer==0.10.2
safetensors==0.4.3
scikit-learn==1.5.1
scipy==1.14.1
six==1.16.0
soupsieve==2.6
stack-data==0.6.3
submitit==1.5.1
sympy==1.13.1
tabulate==0.9.0
threadpoolctl==3.5.0
timm @ git+https://github.com/huggingface/pytorch-image-models.git@730b907b4d45a4713cbc425cbf224c46089fd514
tokenizers==0.15.2
torch==2.4.1
torchaudio==2.4.0
torchvision==0.19.1
tornado==6.4.1
tqdm==4.66.4
traitlets==5.14.3
transformers==4.38.1
triton==3.0.0
typing_extensions==4.12.2
tzdata==2024.1
urllib3==2.2.2
wcwidth==0.2.13
xxhash==3.5.0
yarl==1.9.7

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @msaroufim @ezyang @SherlockNoMad @EikanWang @wenzhe-nrv

**Comment by leslie-fang-intel** [comment]
Hi @Xia-Weiwen, could you help to take a look?
